bundles/org.openhab.binding.bluetooth.bluez/src/main/java/org/openhab/binding/bluetooth/bluez/handler/BlueZBridgeHandler.java

```diff
@@ -155,33 +155,48 @@ public void removeDiscoveryListener(@Nullable BluetoothDiscoveryListener listene
     }
 
     private void startDiscovery() {
+        // we need to make sure the adapter is powered first
+        if (!adapter.getPowered()) {
+            adapter.setPowered(true);
+        }
         if (!adapter.getDiscovering()) {
             adapter.setRssiDiscoveryFilter(-96);
             adapter.startDiscovery();
         }
     }
 
     private void refreshDevices() {
-        logger.debug("Refreshing Bluetooth device list...");
-        List<tinyb.BluetoothDevice> tinybDevices = adapter.getDevices();
-        logger.debug("Found {} Bluetooth devices.", tinybDevices.size());
-        for (tinyb.BluetoothDevice tinybDevice : tinybDevices) {
-            BlueZBluetoothDevice device = getDevice(new BluetoothAddress(tinybDevice.getAddress()));
-            device.updateTinybDevice(tinybDevice);
-            notifyDiscoveryListeners(device);
-        }
-        // clean up orphaned entries
-        synchronized (devices) {
-            for (BlueZBluetoothDevice device : devices.values()) {
-                if (shouldRemove(device)) {
-                    logger.debug("Removing device '{}' due to inactivity", device.getAddress());
-                    device.dispose();
-                    devices.remove(device.getAddress());
+        try {
+            logger.debug("Refreshing Bluetooth device list...");
+            List<tinyb.BluetoothDevice> tinybDevices = adapter.getDevices();
+            logger.debug("Found {} Bluetooth devices.", tinybDevices.size());
+            for (tinyb.BluetoothDevice tinybDevice : tinybDevices) {
+                BlueZBluetoothDevice device = getDevice(new BluetoothAddress(tinybDevice.getAddress()));
+                device.updateTinybDevice(tinybDevice);
+                notifyDiscoveryListeners(device);
+            }
+            // clean up orphaned entries
+            synchronized (devices) {
+                for (BlueZBluetoothDevice device : devices.values()) {
+                    if (shouldRemove(device)) {
+                        logger.debug("Removing device '{}' due to inactivity", device.getAddress());
+                        device.dispose();
+                        devices.remove(device.getAddress());
+                    }
                 }
             }
+            // For whatever reason, bluez will sometimes turn off scanning. So we just make sure it keeps running.
+            startDiscovery();
+            // everything went fine, so lets switch to online
+            updateStatus(ThingStatus.ONLINE);
+        } catch (BluetoothException ex) {
+            String message = ex.getMessage();
```

Comment:
Why is this message parsing necessary? And I guess you need to check if the message is null before calling `lastIndex()`

Reply:
I chose to parse out the most user friendly part of the message since it would be displayed to the user. Otherwise it would look like `tinyb.BluetoothException: GDBus.Error:org.bluez.Error.NotReady: Resource Not Ready`. I'm parsing it so it just comes out to `Resource Not Ready`. But you are correct that I should null check so I will add that.

Reply:
In General I dislike string parsing in front of logging as it is done Always, no matter what log level is set. In this case however it happens only very rarely, therfore I'm fine with it.